bin/migrate_to_single_table.py

```diff
@@ -0,0 +1,319 @@
+                    records = (
+                        self.session.query(IndexRecord)
+                        .order_by(IndexRecord.did)
+                        .limit(batch_size)
+                        .all()
+                    )
+                else:
+                    records = (
+                        self.session.query(IndexRecord)
+                        .order_by(IndexRecord.did)
+                        .filter(IndexRecord.did > last_seen_guid)
+                        .limit(batch_size)
+                        .all()
+                    )
+
+                if not records:
+                    break
+
+                records_to_insert = []
+
+                for record in records:
+                    hashes = self.get_index_record_hash(record.did)
+                    urls = self.get_urls_record(record.did)
+                    url_metadata = self.get_urls_metadata(record.did)
+                    acl = self.get_index_record_ace(record.did)
+                    authz = self.get_index_record_authz(record.did)
+                    alias = self.get_index_record_alias(record.did)
+                    metadata = self.get_index_record_metadata(record.did)
+                    records_to_insert.append(
+                        Record(
+                            guid=record.did,
+                            baseid=record.baseid,
+                            rev=record.rev,
+                            form=record.form,
+                            size=record.size,
+                            created_date=record.created_date,
+                            updated_date=record.updated_date,
+                            content_created_date=record.content_created_date,
+                            content_updated_date=record.content_updated_date,
+                            file_name=record.file_name,
+                            version=record.version,
+                            uploader=record.uploader,
+                            hashes=hashes,
+                            urls=urls,
+                            url_metadata=url_metadata,
+                            acl=acl,
+                            authz=authz,
+                            alias=alias,
+                            record_metadata=metadata,
+                        )
+                    )
+                while records_to_insert:
+                    try:
+                        self.session.bulk_save_objects(records_to_insert)
+                        self.session.commit()
+                        count += len(records_to_insert)
+                        self.logger.info(
+                            f"Done processing {count}/{total_records} records. {(count * 100)/total_records}%"
+                        )
+                        break
+                    except Exception as e:
+                        self.session.rollback()
+                        if "duplicate key value violates unique constraint" in str(e):
```

Comment:
instead of this, you should have an except block for the specific Error type that raises that, you can have multiple except blocks in Python so you can split this. out and keep the general rollback case on Exception too